Add csv output option

kriegerse opened the pull request:
Hello Alexander,

I'm currently working on a monitoring check using your reporting tool. For easier parsing of the reports output I want to introduce a new  `-f`switch supporting the report output as CSV. This will make the report much easier to process by machines. 

```
./BackupPC_report -f csv 
warn_flags,backup_number,host_name,date_time,size_mb,lvl,duratation_min,xfer_err,share_err,file_err,tar_err,age_days
,203,host1,2019-08-12 01:01,3504,1,1,0,0,0,0,0
,209,host2,2019-08-12 01:01,4885,1,3,0,0,0,0,0
E,141,host3,2019-08-12 01:04,11190,1,6,2,0,0,2,0
ED,73,host4,2019-08-12 17:41,86934,1,11,3,0,0,3,0
``` 

Not submitting `-f csv` will keep the current behaviour and print the full formatted report.

Commit:
add option for csv output format

Commit:
remove experimental lib path

Commit:
tidied version

moisseev commented on BackupPC_report:
```diff
@@ -23,10 +23,12 @@ my $bpcUser = 'backuppc';
-GetOptions( "a", "s", "w=f", "help|?", "man" ) || pod2usage(2);
+our ( $opt_a, $opt_f, $opt_s, $opt_w, $opt_help, $opt_man );
+$opt_f = "report";
+GetOptions( "a", "f=s" => "report", "s", "w=f", "help|?", "man" ) || pod2usage(2);
```
Would you clarify what does `=> "report"` do? I cannot find it in the Getopt::Long documentation.

kriegerse replied:
This was definitely not my intention. `=> "report"` should be `=> \$f_opt` for assignment

moisseev replied:
Actually, it is not necessary. By default the variable $opt_X is a destination for the option X. So it can be shortened to
```suggestion
GetOptions( "a", "f=s", "s", "w=f", "help|?", "man" ) || pod2usage(2);
```

kriegerse replied:
okay changed and just using `"f=s"` now

moisseev commented on BackupPC_report:
```diff
@@ -23,10 +23,12 @@ my $bpcUser = 'backuppc';
+GetOptions( "a", "f=s" => "report", "s", "w=f", "help|?", "man" ) || pod2usage(2);
 pod2usage( -exitstatus => 0, -verbose => 2 ) if $opt_man;
 pod2usage(1) if $opt_help;
+pod2usage( -msg => "Unsupported value for -f", -exitstatus => 2 ) if $opt_f !~ /(report|csv)/i;
```
Are you using non-strict checks on purpose? I mean something like `blacsvbla` will match. I'd expect `eq/ne` or anchored regexps .

kriegerse replied:
should be hardened and checks for "allowed" values for `$opt_f`

moisseev commented on BackupPC_report:
```diff
@@ -159,30 +161,45 @@ for my $host ( @ARGV ? @ARGV : @hosts ) {
+    # Print the rest of the table
+    map { printf "%4s [%4d] %-${hostColWidth}s %-16s %6s %3d %1s%5s | %4s %5s %4s %3s | %6d\n", @$_; } @Report;
+}
+elsif ( $opt_f =~ /csv/i ) {
```
What if we move `csv` block before  `report` and for `report` use an else-block without extra condition as `report` is the default?

kriegerse replied:
Also a way, my thought was just to keep clear which format is processed by this section. I'm fine with the else-block as default for the report option

Commit:
- fixed $f_opt assignmend in getopt,
- harden allowed options for $f_opt
- moved default "report" into else-block as suggested

kriegerse commented:
Hi Alexander,
thanks for your inputs. As you see long day and not the best programmer. 
I tried to address them and implement your change request.

Commit:
removed variable assignment for  as this is the default

moisseev commented:
Thank you!

kriegerse commented:
Another short question, is it possible to create a new release for?

moisseev commented:
Sure. Tagged.

kriegerse commented:
Thank you! :-)
https://build.opensuse.org/package/show/home:kriegerse:prod/backuppc-report
